pokemon-checker/src/components/PokemonDisplay/PokemonDisplay.tsx

```diff
@@ -21,31 +21,39 @@ type displayProps = {
 
 export const PokemonDisplay = (props: displayProps) => {
   const [pokeonObject, setPokemonObject] = useState<PokemonDTO>();
-  //TODO (jeremy): Move this factory to a service! Views shouldn't control this.
-  const pokemonFactory: PokemonFactory = new PokemonFactory();
-
-  useEffect(() => {
-    fetchPokemonObject();
-  });
+  // TODO (jeremy): Elminate the factory declaration! The service method is
+  // passed to this component as a prop.
+  const pokemonFactory: PokemonFactory = useMemo(
+    () => new PokemonFactory(),
+    []
+  );
+  /**
+   * Create a PokemonObject from the results retrieved
+   * @param pokemonRetrieved the pokemon retrieved from the API
+   */
+  const createPokemonObject = useCallback(
+    (pokemonRetrieved: IPokemonData): void => {
+      let pokemonToDisplay = pokemonFactory.createPokemon(pokemonRetrieved);
+      setPokemonObject(pokemonToDisplay);
+    },
+    [pokemonFactory]
+  );
 
-  const fetchPokemonObject = () => {
+  // TODO: Eliminate this method. 'getPokemonData' as a Service method
+  // should already return the PokemonDTO - it currently returns raw JSON.
+  const fetchPokemonObject = useCallback(() => {
     const url = props.pokemonUrl;
     if (url && url.length > 0) {
       props
         .getPokemonData(url)
         .then((pokemonRetrieved) => createPokemonObject(pokemonRetrieved))
         .catch(console.log);
     }
-  };
+  }, [props, createPokemonObject]);
 
-  /**
-   * Create a PokemonObject from the results retrieved
-   * @param pokemonRetrieved the pokemon retrieved from the API
-   */
-  const createPokemonObject = (pokemonRetrieved: IPokemonData): void => {
-    let pokemonToDisplay = pokemonFactory.createPokemon(pokemonRetrieved);
-    setPokemonObject(pokemonToDisplay);
-  };
+  useEffect(() => {
+    fetchPokemonObject();
+  }, [fetchPokemonObject]);
```

Comment:
This was the real culprit. We were calling `fetchPokemonObject()` every time the component rendered. It no longer does that, because `fetchPokemonObject` is a dependency - it will only run this code again if it notices that `fetchPokemonObject` has changed.

This does raise a bit of an issue - if `useEffect` is built so that it runs every time a component renders, that means **something** calling `<PokemonDisplay>` is asking it to re-render multiple times per second.

For now, I think this fix should be good enough but there's food for thought on performance going forward.